### PR TITLE
The returned err should be printed

### PR DESCRIPTION
**Pull request by miaoyq**
Signed-off-by: Yanqiang Miao <miao.yanqiang@zte.com.cn>

### CODE REVIEW

**Comment by thaJeztah** [comment]
ping @cpuguy83 looks like you added this in https://github.com/docker/docker/pull/22561 :smile:

**Comment by cpuguy83** [comment]
This is fine, but maybe as a debug message.
And try with `logrus.WithError(err).Debug("First discovery heartbeat failed")`

**Comment by miaoyq** [comment]
@cpuguy83 
Updated

**Review by cpuguy83 — APPROVED** [review]
LGTM

File: daemon/discovery/discovery.go [hunk 1]
```diff
@@ -121,6 +121,8 @@ func (d *daemonDiscoveryReloader) advertiseHeartbeat(address string) {
 	if err := d.initHeartbeat(address); err == nil {
 		ready = true
 		close(d.readyCh)
+	} else {
+		logrus.WithError(err).Debug("First discovery heartbeat failed")
```
Comment (allencloud) [inline daemon/discovery/discovery.go]: Shall we print the detailed error, like:
```
logrus.WithError(err).Debugf("First discovery heartbeat failed: %v", err)
```

Comment (cpuguy83) [inline daemon/discovery/discovery.go]: This would just be the error string?

Comment (miaoyq) [inline daemon/discovery/discovery.go]: @allencloud `WithError(err)` will do it.

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**